execution_engine/src/storage/trie_store/operations/mod.rs

```diff
@@ -252,16 +254,45 @@ where
         if !visited.insert(trie_key) {
             continue;
         }
-        let maybe_retrieved_trie: Option<Trie<K, V>> = store.get(txn, &trie_key)?;
-        match maybe_retrieved_trie {
-            // If we can't find the trie_key; it is missing and we'll return it
+        let maybe_retrieved_trie_bytes = store.get_raw(txn, &trie_key)?;
+        // Optimization: Don't look for descendants of leaves.
```

Comment:
```suggestion
        // Optimization: Don't deserialize leaves.
```

(At least I was confused by this comment at first…)

Reply:
Fixed in [9fcb768](https://github.com/casper-network/casper-node/pull/2844/commits/9fcb76807c46450853f6ca6d04e054031b4426e3).